coderd/database/dbmetrics/dbmetrics.go

```diff
@@ -0,0 +1,1550 @@
+package dbmetrics
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/coder/coder/coderd/database"
+	"github.com/coder/coder/coderd/rbac"
+)
+
+// New returns a database.Store that registers metrics for all queries to reg.
+func New(s database.Store, reg prometheus.Registerer) database.Store {
+	// Don't double-wrap.
+	if _, ok := s.(*metricsStore); ok {
+		return s
+	}
+	queryLatencies := prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: "coderd",
+		Subsystem: "db",
+		Name:      "query_latencies_seconds",
+		Help:      "Latency distribution of queries in seconds.",
+		Buckets:   prometheus.DefBuckets,
+	}, []string{"query"})
+	reg.MustRegister(queryLatencies)
+	return &metricsStore{
+		s:              s,
+		queryLatencies: queryLatencies,
+	}
+}
+
+var _ database.Store = (*metricsStore)(nil)
+
+type metricsStore struct {
+	s              database.Store
+	queryLatencies *prometheus.HistogramVec
+}
+
+func (m metricsStore) Ping(ctx context.Context) (time.Duration, error) {
+	start := time.Now()
+	duration, err := m.s.Ping(ctx)
+	m.queryLatencies.WithLabelValues("Ping").Observe(time.Since(start).Seconds())
+	return duration, err
+}
+
+func (m metricsStore) InTx(f func(database.Store) error, options *sql.TxOptions) error {
+	// No point in measuring this, as it's just a wrapper around the underlying store's InTx method.
```

Comment:
Fair point; I was under the impression that this would be better monitored at the [database side](https://grafana.com/oss/prometheus/exporters/postgres-exporter/assets/postgres_metrics_scrape.txt) but it doesn't appear there's a good metric for that.